### PR TITLE
Mhv 46162 rx update to new mhv endpoint

### PR DESCRIPTION
**Pull request by robertbylight**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production. Will complete the swagger documentation on a separate ticket.

## Summary

- changing the api to call made from va.gov to the new mhv endpoint `/medications` instead of `/gethistoryrx`. decided to extend the prescription model into a new one called prescription_details, updated rspec test and controller functions to reflect new changes.

## Related issue(s)
- [MHV Medications API - Modify the vets-api to expose new MHV Medications API service to VA.gov](https://jira.devops.va.gov/browse/MHV-46162)

## Testing done

- rspec test

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
any areas that use the mhv api model and serializer. 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by mattwrightva — DISMISSED** [review]
Looks good!

**Review by aherzberg — APPROVED** [review]
LGTM (in terms of mobile impact)

**Review by kpethtel — APPROVED** [review]
Looks good. Thanks for bringing this to our attention.

File: modules/my_health/app/serializers/my_health/v1/prescription_details_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      attribute :prescription_source
+      attribute :disclaimer
+      attribute :indication_for_use
+      attribute :indication_for_use_flag
+      attribute :category
+      attribute :tracking_list
+      attribute :rx_rf_records
+      attribute :tracking
+
+      link(:self) { MyHealth::UrlHelper.new.v1_prescription_url(object.prescription_id) }
```
[the remark this answers is not in the document]
Comment (robertbylight) [reply modules/my_health/app/serializers/my_health/v1/prescription_details_serializer.rb]: Hey your right! I went ahead and removed this, thanks!

**Pull request merged**